components/tidb_query_datatype/src/lib.rs

```diff
@@ -1,27 +1,63 @@
-// Copyright 2018 TiKV Project Authors. Licensed under Apache-2.0.
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
 
-//! Representations of types used in the tidb_query component; part of the coprocessor
-//! subsystem.
+//! This crate implements a simple SQL query engine to work with TiDB pushed down executors.
 //!
-//! For an overview of the coprocessor architecture, see the documentation on
-//! [tikv/src/coprocessor](https://github.com/tikv/tikv/blob/master/src/coprocessor/mod.rs).
+//! The query engine is able to scan and understand rows stored by TiDB, run against a
+//! series of executors and then return the execution result. The query engine is provided via
+//! TiKV Coprocessor interface. However standalone UDF functions are also exported and can be used
+//! standalone.
 
-#![cfg_attr(test, feature(test))]
+#![feature(proc_macro_hygiene)]
+#![feature(specialization)]
+#![feature(const_fn)]
+#![feature(iter_order_by)]
+#![feature(test)]
+#![feature(int_error_matching)]
+#![feature(decl_macro)]
+#![feature(str_internals)]
+#![feature(const_loop)]
+#![feature(const_if_match)]
+#![feature(ptr_offset_from)]
+// FIXME: rustc says there are redundant semicolons here but isn't
+// saying where as of nightly-2019-09-05
+// See https://github.com/rust-lang/rust/issues/63967
+#![allow(redundant_semicolon)]
+// FIXME: ditto. probably a result of the above
+#![allow(clippy::no_effect)]
```

Comment:
Could you please tidy them up? I believe not many of them are needed by `datatype`.

Reply:
And same for other new crates